catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/MetacardComparator.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.cache.impl;
+
+import java.util.Set;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.Metacard;
+
+public class MetacardComparator {
+
+    private MetacardComparator() {
+    }
+
+    private static boolean validChecksum(Metacard cachedMetacard, Metacard updatedMetacard) {
+        boolean result = false;
+
+        Attribute cachedMetacardAttr = cachedMetacard.getAttribute(Metacard.CHECKSUM);
+
+        if (cachedMetacardAttr != null) {
```

Comment:
Like that we're making sure that `cachedMetacard` and `updatedMetacard` are `null`, but we probably should do that at the beginning of `isSame()` instead of here. That way we don't need to check that anywhere else.